app/controllers/workorders_controller.rb

```diff
@@ -1,6 +1,7 @@
 class WorkordersController < ApplicationController
   def index
-    @workorders = Workorder.all
+    @workorders = Workorder.where(building_id: session[:current_building_id])
+    @building = Building.find(session[:current_building_id])
```

Comment:
I would switch the order of these two lines and use:
```ruby
@workorders = Workorder.where(building: @building)
```
or try 
```ruby
 @building = Building.find(session[:current_building_id]).includes(: workorders) 
```
to avoid 2 db calls